### ISSUE TITLE
endian.h and compiling for Windows

### ISSUE THREAD

**Issue by CoderBlue** [ponchio/untrunc#40]
I'm trying to compile untrunc for windows. I've been successful in using cygwin to do this but not using mingw-w64 or using cross compilation. I keep getting endian.h errors and when I try to include them or their library I get double declaration errors. Any suggestions ? Thank you.

**Comment by galeksandrp** [comment]
Yes, crossplatform endian.h are exist, but after that you will get undefined uint.

**Comment by anthwlock** [comment]
@Prakash19921206 
iirc you have to choose the pthread threading model when installing cygwin.

**Comment by anthwlock** [comment]
I don't know about cygwin, but I was able to build untrunc on windows using mingw-w64.
On install mingw-w64 asks your preferred threading model:
![mingw_threading_model](https://user-images.githubusercontent.com/39948985/51440084-d5531f00-1cba-11e9-8ef2-7c0d7a7b8652.png)

Eventually you will have more luck cross-compiling using [this branch](https://github.com/anthwlock/untrunc) of untrunc. 
Otherwise your problem seems not to be related to untrunc, but cygwin in general.

**Comment by Prakash19921206** [comment]
@anthwlock, Thanks for your reply.
for this repository, windows build succeeded. after installing all packages in cygwin i installed NASM and YASM and added its path to environment variable(PATH)

then build libav-12.3 using mingw using 
`./configure --enable-pthreads --enable-shared `

and `make`  command using cygwin

finally  using cygwin i used below command
`g++ -o untrunc -I./libav-12.3 file.cpp main.cpp track.cpp atom.cpp mp4.cpp -L./libav-12.3/libavformat -lavformat -L./libav-12.3/libavcodec -lavcodec -L./libav-12.3/libavresample -lavresample -L./libav-12.3/libavutil -lavutil -lpthread  -lz`

now there is a different issue.
not really an issue but missing feature may be! 

this repository takes around 30seconds to fix 5 min video. but your repository takes only couple of seconds (checked on ubuntu but still unable to build your repo on windows)


i think new version is modified, couldn't locate posix or pthreads, but found winpthreads(which is not required i suppose?)
![newui](https://user-images.githubusercontent.com/16747198/51520862-da46d880-1e4a-11e9-8b23-a68e36d5cfc4.PNG)

having other issues in building your repository on windows. continuing the issue there
Thanks & Regards
Prakash

**Comment by anthwlock** [comment]
_Note:_ Automated builds are now available [here](https://github.com/anthwlock/untrunc/releases/tag/master).
